src/services/domain/address.service.ts

```diff
@@ -0,0 +1,43 @@
+import { InferType } from 'yup'
+import prisma from '../../clients/prisma.client'
+import { createAddressValidator } from '../../validations/address.validations'
+
+const create = async (
+  data: InferType<typeof createAddressValidator>
+) => {
+  const customerId = 1
+  await prisma.address.updateMany({
+    where: {
+      customerId
+    },
+    data: {
+      default: false
+    }
+  })
+  return prisma.address.create({
```

Comment:
Wouldn't this throw an error if the ids are duplicate? Or in another case it'd update and create in the same function.

Reply:
Didn't get what you meant, but let me try to explain

One customer can have many addresses, so we would need one of them to be the default one.
For every new address creation, we would need to make all addresses default value "false" and the new one will be the default